gitpod-io/gitpod
# Prebuild trigger checks

## Description

This change introduces the following functionality around activity-based prebuilds:
- Adds a warning for repositories that have received a webhook event for the most recent commit about considering an update to activity-based prebuilds
- Automatically changes prebuild trigger strategy from activity-based to webhook-based when we receive a webhook event
- Automatically changes prebuild trigger strategy from webhook-based to activity-based when creating a workspace and the repository not sending a webhook push event for the last commit

## Related Issue(s)

Fixes ENT-601

## How to test

https://ft-webhook870fa6143a.preview.gitpod-dev.com/workspaces

1. Import a repo
2. Enable prebuilds on it
3. See warning
4. Remove the webhook and push a new commit
5. Observe that the warning disappears
